inc/rule.class.php

```diff
@@ -1595,15 +1601,39 @@ function testCriterias($input, &$check_results) {
    /**
     * Process a criteria of a rule
     *
-    * @param &$criteria  criteria to check
-    * @param &$input     the input data used to check criterias
-   **/
-   function checkCriteria(&$criteria, &$input) {
+    * @param array &$criteria Criteria to check
+    * @param array &$input    The input data used to check criterias
+    * @param array $fields    Contains the fields of the object on which we are
+    *                         applying the rules (ONLY USED for addme_assign and
+    *                         addme_observer special forms)
+    *
+    * @return bool
+    */
+   function checkCriteria(&$criteria, &$input, $fields = []) {
```

Comment:
This function is overrided on seome plugins (at least on `fusioninventory`). Could you test if adding this new parameter (and passing it in every call) can be done without trigerring any PHP NOTICE/WARNING ?

Reply:
It should be better now, I've replaced the new arguments in Rule by a property.